Live-Snap/Live-Snap/Controllers/ExportWallpaperViewController.swift

```diff
@@ -58,24 +77,75 @@ class ExportWallpaperViewController: UIViewController, PHLivePhotoViewDelegate {
             if let livePhoto = livePhoto {
                 self.livePhoto = livePhoto
                 self.livePhotoPreviewView.livePhoto = livePhoto.phLivePhoto
-                self.activityIndicator.stopAnimating()
-                self.showPreview()
+                self.activityIndicatorForCreatingLivePhoto.stopAnimating()
+                self.livePhotoPreviewView.startPlayback(with: .full)
             }
         }
         
     }
     
-    func showPreview() {
-        livePhotoPreviewView.startPlayback(with: .full)
+    func livePhotoView(_ livePhotoView: PHLivePhotoView, willBeginPlaybackWith playbackStyle: PHLivePhotoViewPlaybackStyle) {
+        if livePhotoPreviewFinished {
+            forceTouchNotifierLabel.layer.removeAllAnimations()
+            forceTouchNotifierLabel.removeFromSuperview()
+            userHasForceTouched = true
+        }
+    }
+    
+    override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent?) {
+        if livePhotoPreviewFinished && userHasForceTouched && !view.subviews.contains(toolBar){
+            view.addSubview(toolBar)
+            animateToolBarAndLivePhotoView()
+        }
     }
     
     func livePhotoView(_ livePhotoView: PHLivePhotoView, didEndPlaybackWith playbackStyle: PHLivePhotoViewPlaybackStyle) {
-        if !view.subviews.contains(exportButton) {
-            view.addSubview(exportButton)
-            view.isUserInteractionEnabled = true
+        view.isUserInteractionEnabled = true
+        
+        if !livePhotoPreviewFinished {
+            self.view.addSubview(forceTouchNotifierLabel)
+            UIView.animate(withDuration: 0, delay: 1.0, animations: {
+                self.forceTouchNotifierLabel.center = CGPoint(x: self.view.frame.width/2, y: self.view.frame.height/2)
+            })
+            UIView.animate(withDuration: 1.0, delay: 1.0, options: [.autoreverse, .repeat], animations: {
+                self.forceTouchNotifierLabel.transform = CGAffineTransform(scaleX: 1.2, y: 1.2)
+            }, completion: nil)
+            livePhotoPreviewFinished = true
         }
     }
     
+    func animateToolBarAndLivePhotoView() {
+        let IPhoneXSafeAreaBackgroundView = UIView(frame: CGRect(x: 0, y: self.toolBar.frame.origin.y + 45.0, width: self.view.frame.width, height: 34.0))
+        IPhoneXSafeAreaBackgroundView.backgroundColor = UIColor.snapBlack
+
+        UIView.animate(withDuration: 0.5, animations: {
+            if self.view.isIPhoneX() {
```

Comment:
anything that isn't being animated and just is being added to the screen should be outside this block